### ISSUE TITLE
propertymap configuration should use attributes defined for authomatic.core.User

### ISSUE THREAD

**Issue by ericof** [collective/pas.plugins.authomatic#9]
Authomatic abstracts the differences between providers responses with the authomatic.core.User base attributes.

Currently pas.plugins.authomatic.plugin._make_sheet does the mapping defined in propertymap using the info available on authomatic.core.User.data (and this will vary greatly between providers).

For authomatic.providers.oauth2.Google, the value available in authomatic.core.User.data is : 
```python
{
    u'displayName': u'Erico Andrei',
    u'domain': u'foobar.com',
    u'emails': [{u'type': u'account', u'value': u'erico.andrei@foobar.com'}],
    u'etag': u'"xxxxxxxxxxxx/xxxxxxxxxxxx"',
    u'id': u'198926731687059321180',
    u'image': {u'isDefault': False,
               u'url': u'https://lh3.googleusercontent.com/photo.jpg?sz=50'},
    u'isPlusUser': False,
    u'kind': u'plus#person',
    u'language': u'en_GB',
    u'name': {u'familyName': u'Andrei', u'givenName': u'Erico'},
    u'objectType': u'person',
    u'verified': False
}
```
And if I just want to map email, fullname and get the picture using propertymap, this will not be easily possible.

My suggestion is to define propertymap in the following format:
```javascript
        "propertymap": {
            "email": "email",
            "link": "home_page",
            "locale": "location",
            "name": "fullname",
            "data": {
               "language": "language",
               "domain": "domain",
           }
        },
```
With authomatic.core.User.data being used to get just provider specific information.

p.s.: I volunteer to implement that

**Comment by jensens** [comment]
:+1:  Improvements are very welcome!

The current implementation is very rough, its the classic "make it work for my use case" (which is for new.plone.org github).

I was not aware Google maps it to sub-records like ``u'name': {u'familyName': u'Andrei', u'givenName': u'Erico'},`` - how do you plan to solve this problem?

Btw.: I would not consider this a bug. It works as it is, but limited. So enhancing is fine.

**Comment by jensens** [comment]
Also: Your example above with ``emails`` will not work too, because Plone expects here a string.

**Comment by ericof** [comment]
(I considered it a bug because I was not able to make it run here, but enhancement is better)

I fixed the email on the propertymap (it will use the User.email attribute, that returns a string).

Basically I was not planning to deal with sub-records now (solving my use case as well), but I consider that a challenge to be tackled.

**Comment by jensens** [comment]
closed by #10
